src/arch/xtensa/include/arch/sof.h

```diff
@@ -69,7 +71,7 @@ static inline void fill_core_dump(struct sof_ipc_dsp_oops_xtensa *oops,
 	oops->plat_hdr.configidlo = 0;
 #endif
 	oops->plat_hdr.numaregs = XCHAL_NUM_AREGS;
-	oops->plat_hdr.stackoffset = ((void *)&oops->stack) - (void *)oops;
+	oops->plat_hdr.stackoffset = oops_size;
```

Comment:
@jajanusz Isn't the panic dump written after oops and this sets stackoffset to the same place as where the panic will get written (in panic_rewind())? So if I use this to read the stack, I'll be reading from the wrong place?

Reply:
@kv2019i You are right, that's why I commented that it's weird that it worked at all. This field makes no sense for me, I just directly ported it from previous version.

Reply:
@kv2019i I'll do some cleanup here. On the kernel side  you can use `stackptr` field, this one works, `stack` and `stackoffset` look like some mistakes.

Reply:
@jajanusz I think I can get the stack offset by jumpin over totalsize+panicsize and then I'm at start of stack. The "stackptr" does not seem useful as it prints the stack address in FW address space (while I need to get an offset to the stack within the dumped range in sram). So in that sense, "stackoffset" would seem useful if the value would be correct and take panic section into account. Not sure about "stackptr"...

Reply:
@kv2019i  `stackptr` is used by coredumper so it should be here. I can fix stackoffset tho.